src/net/tcp/socket.rs

```diff
@@ -334,6 +334,20 @@ impl Drop for TcpSocket {
     }
 }
 
+impl From<&TcpStream> for TcpSocket {
+    fn from(stream: &TcpStream) -> TcpSocket {
```

Comment:
If drop the `TcpSocket` now it will close the `TcpStream`'s file descriptor. Can't use `From<&TcpStream>`.

Reply:
This could work with something like socket2's `SockRef`: https://github.com/rust-lang/socket2/blob/9c4a404d5e44df1ecc56d62d35ee22109a886b5f/src/sockref.rs#L59. Maybe it's best to wait for #1381, which I'll work on (likely) next year.